File: 1380. Lucky Numbers in a Matrix/luckyNumbers.js
/**
 * @param {number[][]} matrix
 * @return {number[]}
 */
const luckyNumbers  = (matrix) => {
  if (!matrix.length) {
    return null;
  }

  const ROWS = matrix.length;
  const COLS = matrix[0].length;

  const res = [];
  const maxInRows = new Set();
  for (let r = 0; r < ROWS; r++) {
    maxInRows.add(Math.min(...matrix[r]));
  }

  const minInCols = new Set(); 
  for (let c = 0; c < COLS; c++) {
    let currMax = matrix[0][c];
    for (let r = 0; r < ROWS; r++) {
      currMax = Math.max(currMax, matrix[r][c]);
    }
    minInCols.add(currMax);
  }

  for (let n of maxInRows) {
    if (minInCols.has(n)) {
      res.push(n);
    }
  }

  return res;
};


const matrix = [
  [3, 7, 8],
  [9, 11, 13],
  [15, 16, 17]
];
console.log('luckyNumbers: ', luckyNumbers(matrix));

File: 1380. Lucky Numbers in a Matrix/luckyNumbers.py
from typing import List

class Solution:
  def luckyNumbers (self, matrix: List[List[int]]) -> List[int]:
    if not matrix:
      return None
    
    ROWS = len(matrix)
    COLS = len(matrix[0])
    res = []
    minInRow = set()

    for r in range(ROWS):
      minInRow.add(min(matrix[r]))

    maxInCol = set()
    for c in range(COLS):
      currMax = matrix[0][c]
      for r in range(ROWS):
        currMax = max(currMax, matrix[r][c])
      maxInCol.add(currMax)

    for n in minInRow:
      if n in maxInCol:
        res.append(n) 

    return res


matrix = [
  [3, 7, 8],
  [9, 11, 13],
  [15, 16, 17]
]
solution = Solution()
print('luckyNumbers: ', solution.luckyNumbers(matrix))
